Engine/Source/EmitterShape.cpp

```diff
@@ -0,0 +1,144 @@
+#include "EmitterShape.h"
+
+# define M_PI   3.14159265358979323846f
+
+EmitterShape::EmitterShape()
+{
+    mType = Type::NONE;
+	mShapeRadius = 1.0f;
+	mShapeAngle = 3.1415f / 4.0f;
+	mShapeSize = float2(1.0f, 1.0f);
+}
+
+EmitterShape::~EmitterShape()
+{
+}
+
+void EmitterShape::Save(Archive& archive) const
+{
+    archive.AddInt("ShapeType", mType);
+    archive.AddFloat("ShapeRadius", mShapeRadius);
+    archive.AddFloat("ShapeAngle", mShapeAngle);
+    archive.AddFloat2("ShapeSize", mShapeSize);
+}
+
+void EmitterShape::LoadFromJSON(const rapidjson::Value& data)
+{
+    if (data.HasMember("ShapeRadius") && data["ShapeRadius"].IsFloat())
+    {
+        mShapeRadius = data["ShapeRadius"].GetFloat();
+    }
+    if (data.HasMember("ShapeAngle") && data["ShapeAngle"].IsFloat())
+    {
+        mShapeAngle = data["ShapeAngle"].GetFloat();
+    }
+    if (data.HasMember("ShapeSize") && data["ShapeSize"].IsArray())
+    {
+        const rapidjson::Value& values = data["ShapeSize"];
+        float x{ 0.0f }, y{ 0.0f };
+        if (values.Size() == 2 && values[0].IsFloat() && values[1].IsFloat())
+        {
+            x = values[0].GetFloat();
+            y = values[1].GetFloat();
+        }
+
+        mShapeSize = float2(x, y);
+    }
+
+}
+
+void EmitterShape::CopyShape(const EmitterShape& original)
+{
+    mShapeAngle = original.mShapeAngle;
+    mShapeRadius = original.mShapeRadius;
+    mShapeSize = original.mShapeSize;
+}
```

Comment:
This can be a copy constructor, no?

Reply:
We use inherited classes for the different shapes so that would mean checking which constructor to call. Right now this is faster and easier, but I'm realizing that there isn't much benefit in having different classes. I'll have to discuss it with Marc.